### PR TITLE
test: swap details dropdown unit test

### PR DESCRIPTION
**Pull request by lynnshaoyu**
add jest unit tests with complete coverage for SwapDetailsDropdown.tsx component

### CODE REVIEW

File: src/setupTests.ts [hunk 1]
```diff
@@ -10,6 +10,8 @@ if (typeof global.TextEncoder === 'undefined') {
   global.TextDecoder = TextDecoder as typeof global.TextDecoder
 }
 
+global.ResizeObserver = require('resize-observer-polyfill')
```
Comment (zzmp) [inline src/setupTests.ts:13]: This should be explicitly added to devDependencies. Right now it's just included as a sub-dep of widgets.

Should we be including this in the build? Or can we always depend on resizeObserver being defined?

**Review by zzmp — CHANGES_REQUESTED** [review]

File: src/setupTests.ts (hunk 1 above)
Comment (lynnshaoyu) [reply src/setupTests.ts:13]: I added it to dev dependencies but we still need this line here for the tests to pass.

Comment (zzmp) [reply src/setupTests.ts:13]: that's expected - it's because jest is running its own DOM environment so it's not actually an up-to-date browser

**Review by JFrankfurt — APPROVED** [review]

File: src/setupTests.ts (hunk 1 above)
Comment (zzmp) [reply src/setupTests.ts:13]: You shouldn't require unless you need to do so conditionally. Split this into two lines:

```
import ResizeObserver from 'resize-observer-polyfill'
...
global.ResizeObserver = ResizeObserver
```

File: src/components/swap/SwapDetailsDropdown.test.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+jest.mock('@web3-react/core', () => {
+  const web3React = jest.requireActual('@web3-react/core')
+  return {
+    ...web3React,
+    useWeb3React: () => ({
+      chainId: 1,
+    }),
+  }
+})
+
```
Comment (zzmp) [inline src/components/swap/SwapDetailsDropdown.test.tsx:15]: This is no longer necessary - it is mocked in `setupTests`.
You'll need to remock it (as `mocked(useWeb3React).mockReturnValue({...})`), but only if you want to mock a connected account.

File: src/components/swap/SwapDetailsDropdown.test.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+describe('SwapDetailsDropdown.tsx', () => {
+  it('matches base snapshot', () => {
```
Comment (zzmp) [inline src/components/swap/SwapDetailsDropdown.test.tsx:17]:
```suggestion
  it('renders a trade', () => {
```

File: src/components/swap/SwapDetailsDropdown.test.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        trade={TEST_TRADE_EXACT_INPUT}
+        syncing={false}
+        loading={false}
+        allowedSlippage={TEST_ALLOWED_SLIPPAGE}
+      />
+    )
+    expect(asFragment()).toMatchSnapshot()
+  })
+
+  it('loading state contains expected elements', () => {
```
Comment (zzmp) [inline src/components/swap/SwapDetailsDropdown.test.tsx:29]:
```suggestion
  it('renders loading state', () => {
```

File: src/components/swap/SwapDetailsDropdown.test.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    render(
+      <SwapDetailsDropdown trade={undefined} syncing={true} loading={true} allowedSlippage={TEST_ALLOWED_SLIPPAGE} />
+    )
+    expect(screen.getByText('Fetching best price...')).toBeInTheDocument()
+  })
+
+  it('interactive components work as expected once trade is loaded', () => {
```
Comment (zzmp) [inline src/components/swap/SwapDetailsDropdown.test.tsx:36]:
```suggestion
  it('is interactive once loaded', () => {
```

**Review by zzmp — CHANGES_REQUESTED** [review]

File: src/components/swap/SwapDetailsDropdown.test.tsx (hunk 2 above)
Comment (lynnshaoyu) [reply src/components/swap/SwapDetailsDropdown.test.tsx]: ty done

File: src/components/swap/SwapDetailsDropdown.test.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      <SwapDetailsDropdown
+        trade={TEST_TRADE_EXACT_INPUT}
+        syncing={false}
+        loading={false}
+        allowedSlippage={TEST_ALLOWED_SLIPPAGE}
+      />
+    )
+    expect(screen.getByTestId('swap-details-header-row')).toBeInTheDocument()
+    expect(screen.getByTestId('trade-price-container')).toBeInTheDocument()
+    fireEvent.click(screen.getByTestId('swap-details-header-row'))
```
Comment (zzmp) [inline src/components/swap/SwapDetailsDropdown.test.tsx:38]: Should this be a userEvent?

**Review by zzmp — CHANGES_REQUESTED** [review]

File: src/components/swap/SwapDetailsDropdown.test.tsx (hunk 6 above)
Comment (lynnshaoyu) [reply src/components/swap/SwapDetailsDropdown.test.tsx]: fixed

File: src/components/swap/SwapDetailsDropdown.test.tsx [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+      <SwapDetailsDropdown
+        trade={TEST_TRADE_EXACT_INPUT}
+        syncing={false}
+        loading={false}
+        allowedSlippage={TEST_ALLOWED_SLIPPAGE}
+      />
+    )
+    expect(screen.getByTestId('swap-details-header-row')).toBeInTheDocument()
+    expect(screen.getByTestId('trade-price-container')).toBeInTheDocument()
+    userEvent.click(screen.getByTestId('swap-details-header-row'))
```
Comment (zzmp) [inline src/components/swap/SwapDetailsDropdown.test.tsx:39]: This should be wrapped in `act`

```suggestion
    await act(() => userEvent.click(screen.getByTestId('swap-details-header-row')))
```

**Review by zzmp — CHANGES_REQUESTED** [review]

**Review by TDG510 — APPROVED** [review]

**Review by zzmp — CHANGES_REQUESTED** [review]
split the import and then lgtm

**Review by zzmp — APPROVED** [review]

**Pull request merged**